lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -91,8 +94,8 @@ defmodule Mix.Compilers.Elixir do
   end
 
   defp mtimes_and_sizes(sources) do
-    Enum.reduce(sources, %{}, fn source(source: source, external: external), map ->
-      Enum.reduce([source | external], map, fn file, map ->
+    Enum.reduce(sources, %{}, fn(source(source: source, external: external), map) ->
```

Comment:
Please don't change the syntax of unrelated functions while doing those changes. :)